partner_multi_relation/models/res_partner_relation.py

```diff
@@ -154,3 +164,32 @@ def _check_relation_uniqueness(self):
                     _('There is already a similar relation with '
                       'overlapping dates')
                 )
+
+    @api.multi
+    def check_is_unique(self, vals):
+        type_id = self.env['res.partner.relation.type']
+        for line in self.env['res.partner.relation'].search([]):
+            if line.type_id.is_unique:
+                count = 0
+                if 'right_partner_id' in vals:
+                    if vals.get('right_partner_id') == \
+                       line.right_partner_id.id:
+                        count += 1
+                else:
+                    if self.right_partner_id == line.right_partner_id.id:
+                        count += 1
+                if 'type_id' in vals:
+                    if type_id.browse(vals.get('type_id')).name == \
+                       line.type_id.name:
+                        count += 1
+                    if (type_id.browse(vals.get('type_id')).is_unique
+                       or line.type_id.is_unique):
+                        count += 1
+                else:
+                    if self.type_id.name == line.type_id.name:
+                        count += 1
+                    if self.type_id.is_unique or line.type_id.is_unique:
+                        count += 1
+                if count == 3:
+                    return False
```

Comment:
@osi-scampbell Why don't you just search if the same relation already exists?

Something like:
```
right = vals['right_partner_id'] or self.right_partner_id
type = vals['type_id] or self.type_id
left = vals['left_partner_id'] or self.left_partner_id
res = self.search([('left_partner_id', '=', left), 'type_id', '=', type), 'right_partner_id', '=', right)])
...
```